napari/_vispy/vispy_points_layer.py

```diff
@@ -100,14 +98,13 @@ def _on_highlight_change(self):
             if data.ndim == 1:
                 data = np.expand_dims(data, axis=0)
             size = self.layer._sizes_view[self.layer._highlight_index]
-            face_color = [
-                self.layer.face_colors[i]
-                for i in self.layer._indices_view[self.layer._highlight_index]
+            face_color = self.layer.face_color[
+                self.layer._indices_view[self.layer._highlight_index]
```

Comment:
This was throwing errors for me when I tried selecting points. This line of code is to grab the face colors for the selected points. Thus, we want to get the `self.layers.face_colors` corresponding with the indices of the highlighted points (`self.layer._highlight_index`). I am short on time, but I couldn't get the indices to broadcast onto the `ColorArray` properly, so did the following (feel free to use or modify)

```python
            # get the colors for each highlighted point
            face_color = [
                 self.layer.face_colors[i]
                 for i in self.layer._indices_view[self.layer._highlight_index]
            ]
            # use your nice function transform the list to a color array
            face_color = transform_color(face_color)
```

Reply:
OHAI! Here I was thinking my concerns about `__getitem__` behaviour were purely theoretical... ;)